core/src/main/java/com/netflix/conductor/core/utils/Lock.java

```diff
@@ -0,0 +1,48 @@
+package com.netflix.conductor.core.utils;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface implemented by a distributed lock client.
+ *
+ * A typical usage:
+ *   if (acquireLock(workflowId, 5, TimeUnit.MILLISECONDS)) {
+ *      [load and execute workflow....]
+ *      ExecutionDAO.updateWorkflow(workflow);  //use optimistic locking
+ *   } finally {
+ *     releaseLock(workflowId)
+ *   }
+ *
+ */
+
+public interface Lock {
+
+    /**
+     * acquires a re-entrant lock on lockId, blocks indefinitely on lockId until it succeeds
+     * @param lockId resource to lock on
+     */
+    void acquireLock(String lockId);
+
+    /**
+     * acquires a re-entrant lock on lockId, blocks for timeToTry duration before giving up
+     * @param lockId resource to lock on
+     * @param timeToTry blocks up to timeToTry duration in attempt to acquire the lock
+     * @param unit time unit
+     * @return
+     */
+    boolean acquireLock(String lockId, long timeToTry, TimeUnit unit);
+
+    /**
+     * acquires a re-entrant lock on lockId, blocks for timeToTry duration before giving up
```

Comment:
Consider adding a description about usage of `leaseTIme` to the javadoc. The difference between this API and the one above is not clear.

Reply:
Maybe rename the first one to waitForLock?